client/my-sites/stats/stats-module/index.jsx

```diff
@@ -126,10 +111,10 @@ class StatsModule extends Component {
 			useShortLabel,
 		} = this.props;
 
-		const noData = data && this.state.loaded && ! data.length;
+		const noData = data && ! requesting && ! data.length;
 
 		// Only show loading indicators when nothing is in state tree, and request in-flight
-		const isLoading = ! this.state.loaded && ! ( data && data.length );
+		const isLoading = requesting && ! ( data && data.length );
```

Comment:
There is a behavior change here, affecting the initial render and when the `query` changes. There are three renders:
1. initially, `requesting` is `false` and `data` is `[]`. Therefore `noData` is `true` and "No stuff recorded" label is shown.
2. only after mount the request is triggered (in effect) and `requesting` becomes `true`, rerendering to a placeholder UI
3. after the request finishes, `requesting` is back to `false` and data are displayed.

The first render with `requesting === false` is undesired and the original code with `this.state.loaded` didn't do it.

I'm not sure how much of a problem this is, but the original behavior can be easily kept by merely switching to `componentDidUpdate`.

Reply:
Changed to `componentDidUpdate()` with some lint rule disabling, curious to hear your thoughts @jsnajdr.